### ISSUE TITLE
Switch multi-GPU training to DistributedDataParallel

### ISSUE THREAD

**Issue by eelcovdw** [Roxot/AEVNMT.pt#26]
The current implementation of distributed training is done with DataParallel, which has some drawbacks.

- Pytorch is switching to DistributedDataParallel which is much better, but requires us to rewrite most training code (and possibly some model parts).
- DataParallel is not compatible with RNN cells, and can give worse results with the Transformer (due to normalization?).
- My suggestion: Rewrite training code (Possibly with Pytorch-lighting), build it around DistributedDataParallel.
